From: Wang Shilong <wangsl.fnst@cn.fujitsu.com>
Date: Wed, 15 Jan 2014 00:26:43 +0800
Patch-mainline: 3.14
Git-commit: ffcfaf81795471be3c07d6e3143bff31edca5d5a
Subject: [PATCH] Btrfs: fix wrong search path initialization before
 searching tree root

To search tree root without transaction protection, we should neither search commit
root nor skip locking here, fix it.

Signed-off-by: Wang Shilong <wangsl.fnst@cn.fujitsu.com>
Signed-off-by: Josef Bacik <jbacik@fb.com>
Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: David Sterba <dsterba@suse.cz>
---
 fs/btrfs/send.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/fs/btrfs/send.c
+++ b/fs/btrfs/send.c
@@ -2139,7 +2139,7 @@ static int send_subvol_begin(struct send
 	char *name = NULL;
 	int namelen;
 
-	path = alloc_path_for_send();
+	path = btrfs_alloc_path();
 	if (!path)
 		return -ENOMEM;
 
